flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -261,6 +267,57 @@ private boolean isStreamingSource() {
 		return new DataStreamSource<>(source);
 	}
 
+	private DataStream<RowData> createStreamSourceForNonPartitionTable(
+			StreamExecutionEnvironment execEnv,
+			TypeInformation<RowData> typeInfo,
+			HiveTableInputFormat inputFormat,
+			HiveTablePartition hiveTable) {
+		HiveTableFileInputFormat fileInputFormat = new HiveTableFileInputFormat(
+				inputFormat,
+				hiveTable);
+		fileInputFormat.setFilePath(getFilePath());
```

Comment:
Remove `getFilePath`, we can get path from `allHivePartitions`(should be a single object when no partitions.).